### ISSUE TITLE
empty space in dimnames breaks spread_draws/gather_draws

### ISSUE THREAD

**Issue by Schmidtpk** [mjskay/tidybayes#279]
If dimnames in an mcmc.list have empty spaces, the code breaks with 

> "Error: Each row of output must be identified by a unique combination of keys. [...]"

This happens for example when using nimble, which has dimnames 

> varname[1, 1] 

instead of 

> varname[1,1]`

Of course, users can just manipulate the names before using tidybayes with

```
for(i in 1:length(mcmc.list))
  dimnames(mcmc.list[[i]])[[2]]<-gsub(' ','',dimnames(mcmc.list[[i]])[[2]])
```

but a change in tidybayes would make it much easier.

**Comment by mjskay** [comment]
You should be able to handle this using the `sep` argument to `spread_draws()` / `gather_draws()`. Something like `spread_draws(..., sep = ", ")` should do it, or if the presence of a space is inconsistent (if you sometimes have none or sometimes have more than one) you could do something like `spread_draws(..., sep = ", *")`

It does raise the question if the default value of `sep` should be changed. The current default is `"[, ]"`, i.e., one comma or one space. Could change to something like `"[, ] *"`, i.e. one comma or space optionally followed by any number of spaces. I doubt that would break anyone's existing code.

**Comment by Schmidtpk** [comment]
> Something like spread_draws(..., sep = ", ")

Thanks. This works fine; same for the general version.

Maybe an alternative would be to throw a suggestive error or warning message in case the default change is deemed to risky. At least for me it took some time to figure out that the difference in dimnames was what caused the problem.

Also, nimble could possibly just adhere to the standards. I'll post an issue there, too.

**Comment by mjskay** [comment]
Good point that the error when a spec breaks could suggest use of `sep`.

TODOs for my benefit:

- [ ] consider changing default of `sep` arg to something like `"[, ] *"`
- [ ] add a note in errors in spread_draws suggesting people look at `sep` if it seems appropriate
